src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -0,0 +1,94 @@
+#include "generate_pruning_statistics.hpp"
+
+#include <atomic>
+#include <iostream>
+#include <thread>
+#include <unordered_set>
+
+#include "operators/table_wrapper.hpp"
+#include "resolve_type.hpp"
+#include "statistics/column_statistics.hpp"
+#include "statistics/statistics_objects/equal_distinct_count_histogram.hpp"
+#include "statistics/statistics_objects/generic_histogram_builder.hpp"
+#include "statistics/statistics_objects/min_max_filter.hpp"
+#include "statistics/statistics_objects/null_value_ratio_statistics.hpp"
+#include "statistics/statistics_objects/range_filter.hpp"
+#include "statistics/table_statistics.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "storage/table.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+template <typename T>
+void create_pruning_filter_for_segment(ColumnStatistics<T>& column_statistics, const pmr_vector<T>& dictionary) {
```

Comment:
I know what you mean. Chunks can have a ColumnStatistics object for each of their Segments to be used as pruning statistics. Since statistics objects can be used for both pruning (which happens on a single Chunk/Segment) and cardinality estimation (which happens on the entire Table/column) ColumnStatistics is used to wrap the statistics of both purposes.

I once called this "VerticalStatisticsSlice" as a name that covers both purposes equally, but I don't think that's any better

Reply:
Right now I feel like renaming `ColumnStatistics` to `AttributeStatistics` with the class comment "// Represents either an entire Column or a single Segment` is the best course. You cool with this?

Reply:
Sounds good.

Reply:
Renamed. Variables of this type are called "column_statistics" in cardinality estimation contexts and "segment_statistics" in chunk pruning contexts